### PR TITLE
SEAB-5762: Pagination for entry versions

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
This PR implements pagination for versions in both the My workflows and the public entry page.

Pagination for workflow with 600+ versions
image

My workflows page (Note the hidden versions are visible)
image

Public page (Hidden versions are not visible)
image

**Review Instructions**
Go to https://qa.dockstore.org/workflows/github.com/broadinstitute/gatk/cnv_somatic_pair_workflow:4.2.0.0?tab=versions
and verify pagination works

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5762

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

File: src/app/workflow/versions/versions-datasource.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        .getPublicWorkflowVersions(workflowId, pageSize, pageIndex, sortCol, sortDirection, 'response')
+        .pipe(finalize(() => this.loadingSubject$.next(false)))
+        .subscribe((versions) => {
+          this.versionsSubject$.next(versions.body);
+          this.versionsLengthSubject$.next(Number(versions.headers.get('X-total-count')));
+        });
+    } else {
+      this.workflowsService
+        .getWorkflowVersions(workflowId, pageSize, pageIndex, sortCol, sortDirection, 'response')
+        .pipe(finalize(() => this.loadingSubject$.next(false)))
```
Comment (coverbeck) [inline src/app/workflow/versions/versions-datasource.ts:60]: You should try avoid this duplication. One way would be to get an appropriate Observable based on `publicPage` around line 49, then you'll only need the pipe/subscribe code once.

File: src/app/workflow/versions/versions.component.ts [hunk 2]
```diff
@@ -148,20 +192,41 @@ export class VersionsWorkflowComponent extends Versions implements OnInit, OnCha
...
+      publicPage,
+      this.workflowId,
+      direction,
+      this.paginator.pageIndex * this.paginator.pageSize,
+      this.paginator.pageSize,
+      this.sortCol
+    );
   }
 
+  ngOnDestroy() {
```
Comment (coverbeck) [inline src/app/workflow/versions/versions.component.ts:226]: I don't think you need this method. There's already an `ngOnDestroy` implementation in entry-tab.ts that's getting duplicated here.

**Review by coverbeck — COMMENTED** [review]
I ran this locally and it's working well for me. A few nit comments...

Also, if you haven't already, can you create a ticket to remove workflow versions from the published endpoint response; or if not remove, at least make it required via a query param to fetch them. With your changes, you can see all workflows versions, but we're also potentially fetching 200 workflow versions twice -- we should only be relying on pagination.

**Review by svonworl — APPROVED** [review]

**Review by denis-yuen — COMMENTED** [review]
Looks good in principle (screenshots), will follow feedback conversation

File: .circleci/config.yml [hunk 3]
```diff
@@ -197,7 +197,7 @@ jobs:
       - setup_integration_test
       - run:
           name: Test
-          command: bash -i -c 'npx cypress run --browser chrome --record --config numTestsKeptInMemory=1 --reporter junit --spec cypress/e2e/<< parameters.integration_test_name >>/**/*'
+          command: bash -i -c 'npx cypress run --record --config numTestsKeptInMemory=1 --reporter junit --spec cypress/e2e/<< parameters.integration_test_name >>/**/*'
```
[the remark this answers is not in the document]
Comment (hyunnaye) [reply .circleci/config.yml:200]: This was the cherry pick from Kathy to fix the flaky error

**Comment by hyunnaye** [comment]
> at least make it required via a query param to fetch them

Ticket: https://ucsc-cgl.atlassian.net/browse/SEAB-6822

**Review by coverbeck — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Pull request merged**